src/sdks/core/src/cpp/sdk/cpptest/unit/accessibilityTest.cpp

```diff
@@ -0,0 +1,370 @@
+            ADD_FAILURE() << "Expected styles.fontFamily to have a value";
+
+        if (styles.fontOpacity.has_value())
+            EXPECT_EQ(styles.fontOpacity.value(), expectedValues["styles"]["fontOpacity"]);
+        else
+            ADD_FAILURE() << "Expected styles.fontOpacity to have a value";
+
+        if (styles.fontSize.has_value())
+            EXPECT_EQ(styles.fontSize.value(), expectedValues["styles"]["fontSize"]);
+        else
+            ADD_FAILURE() << "Expected styles.fontSize to have a value";
+
+        if (styles.textAlign.has_value())
+            EXPECT_EQ(styles.textAlign.value(), expectedValues["styles"]["textAlign"]);
+        else
+            ADD_FAILURE() << "Expected styles.textAlign to have a value";
+
+        if (styles.textAlignVertical.has_value())
+            EXPECT_EQ(styles.textAlignVertical.value(), expectedValues["styles"]["textAlignVertical"]);
+        else
+            ADD_FAILURE() << "Expected styles.textAlignVertical to have a value";
+
+        if (styles.windowColor.has_value())
+            EXPECT_EQ(styles.windowColor.value(), expectedValues["styles"]["windowColor"]);
+        else
+            ADD_FAILURE() << "Expected styles.windowColor to have a value";
+
+        if (styles.windowOpacity.has_value())
+            EXPECT_EQ(styles.windowOpacity.value(), expectedValues["styles"]["windowOpacity"]);
+        else
+            ADD_FAILURE() << "Expected styles.windowOpacity to have a value";
+    }
+    else 
+    {
+        ADD_FAILURE() << "Expected closedCaptionSettings.styles to have a value";
+    }
+ 
+    EXPECT_EQ(closedCaptionSettings.preferredLanguages.value()[0], expectedValues["preferredLanguages"][0]);
+    EXPECT_EQ(closedCaptionSettings.preferredLanguages.value()[1], expectedValues["preferredLanguages"][1]);
+}
+
+TEST_F(AccessibilityTest, VoiceGuidance)
+{
+    nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Accessibility.voiceGuidance"));
+
+    auto voiceGuidance = Firebolt::IFireboltAccessor::Instance().AccessibilityInterface().voiceGuidance(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve voiceGuidance from Accessibility.voiceGuidance() method";
+
+    EXPECT_EQ(voiceGuidance.enabled, expectedValues["enabled"]);
+    EXPECT_EQ(voiceGuidance.speed, expectedValues["speed"]);
+}
+
+TEST_F(AccessibilityTest, VoiceGuidanceSettings)
+{
+    nlohmann::json_abi_v3_11_3::json expectedValues = nlohmann::json::parse(jsonEngine->get_value("Accessibility.voiceGuidanceSettings"));
+
+    auto voiceGuidanceSettings = Firebolt::IFireboltAccessor::Instance().AccessibilityInterface().voiceGuidanceSettings(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve voiceGuidanceSettings from Accessibility.voiceGuidanceSettings() method";
+
+    EXPECT_EQ(voiceGuidanceSettings.enabled, expectedValues["enabled"]);
+    EXPECT_EQ(voiceGuidanceSettings.speed, expectedValues["speed"]);
```

Comment:
The voice guidance tests need to capture the new fields, `rate` and `navigationHints`.